src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -248,10 +245,71 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+        // and propagate those changes to the UI
+        _ViewModelChangedRevoker = _State.Profile().PropertyChanged(winrt::auto_revoke, [weakThis{ get_weak() }](auto&&, const PropertyChangedEventArgs& args) {
+            if (auto page{ weakThis.get() })
+            {
+                if (args.PropertyName() == L"CursorShape")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentCursorShape" });
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"IsVintageCursor" });
+                }
+                else if (args.PropertyName() == L"BackgroundImageStretchMode")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentBackgroundImageStretchMode" });
+                }
+                else if (args.PropertyName() == L"AntialiasingMode")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentAntiAliasingMode" });
+                }
+                else if (args.PropertyName() == L"CloseOnExit")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentCloseOnExitMode" });
+                }
+                else if (args.PropertyName() == L"BellStyle")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentBellStyle" });
+                }
+                else if (args.PropertyName() == L"ScrollState")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentScrollState" });
+                }
+                else if (args.PropertyName() == L"FontWeight")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentFontWeight" });
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"IsCustomFontWeight" });
+                }
+                else if (args.PropertyName() == L"ColorSchemeName")
+                {
+                    page->_PropertyChangedHandlers(*page, PropertyChangedEventArgs{ L"CurrentColorScheme" });
+                }
+                else if (args.PropertyName() == L"BackgroundImageAlignment")
+                {
+                    // reset all of the buttons to unchecked.
+                    // set the one for the value we actually have.
+                    const auto alignment{ static_cast<int32_t>(page->_State.Profile().BackgroundImageAlignment()) };
+                    for (const auto& biButton : page->_BIAlignmentButtons)
+                    {
+                        if (const auto& biButtonAlignment{ biButton.Tag().try_as<int32_t>() })
+                        {
+                            biButton.IsChecked(biButtonAlignment == alignment);
+                        }
+                    }
```

Comment:
is there a helper function for this? we do it in 2 places i think

Reply:
We do a very similar thing in the other one (there we do `button == biButton` instead). I'll pull this out into a helper function though, you're right.